src/App.tsx

```diff
@@ -1,24 +1,173 @@
+    if (titleTodo) {
+      setTempTodo(true);
+      try {
+        await postTodo(
+          { title: titleTodo, userId: USER_ID, completed: false },
+        );
+
+        getListTodo();
+        setTitleTodo('');
+        setTempTodo(false);
+      } catch (e) {
+        setError('Oops, something were wrong, please try again later');
+      }
+    }
+  };
+
+  const onChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleTodo(event.target.value);
+  };
+
+  const deleteTodoHandler = (id: number) => {
+    deleteTodo(id).then(() => getListTodo());
+  };
+
+  const deleteAllCompleteTodo = () => {
+    listTodo.forEach((todo: Todo) => {
+      if (todo.completed) {
+        deleteTodoHandler(todo.id);
+      }
+    });
+  };
+
+  const renderClearButton = useMemo(() => {
+    if (listTodo.some((todo: Todo) => todo.completed)) {
+      return (
+        <button
+          type="button"
+          className="todoapp__clear-completed"
+          onClick={deleteAllCompleteTodo}
+        >
+          Clear completed
+        </button>
+      );
+    }
+
+    return <></>;
+  }, [listTodo]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className="todoapp__toggle-all active"
+          />
+          <form onSubmit={onSubmitHandler}>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={titleTodo}
+              onChange={onChangeHandler}
+              disabled={tempTodo}
+            />
+          </form>
+        </header>
+
+        <section className="todoapp__main">
+          {
+            listTodo.map((todo: Todo) => (
+              <Item
+                todo={todo}
+                key={todo.id}
+                deleteTodo={deleteTodoHandler}
+                updateTodos={getListTodo}
+              />
+            ))
+          }
+          {tempTodo && (
+            <Item
+              todo={{
+                id: +uuid(),
+                userId: USER_ID,
+                title: titleTodo,
+                completed: false,
+              }}
+              temp
+            />
+          )}
+        </section>
+
+        {showFooter ? (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${length} items left`}
```

Comment:
i think we can manually calculate the length here instead of using useState

Reply:
i try, but how i wrote down, in second useEffect i count all elements, i can create async function and add in html, but why if i can just in one place count all elements and check need render footer or no